Lib/importlib/_bootstrap_external.py

```diff
@@ -1331,8 +1331,10 @@ def _get_spec(cls, fullname, path, target=None):
         #  the list of paths that will become its __path__
         namespace_path = []
         for entry in path:
-            if not isinstance(entry, (str, bytes)):
+            if not (isinstance(entry, (str, bytes)) or
+                    hasattr(entry, '__fspath__')):
```

Comment:
It seems that this change doesn't take the comment on the old PR into account.

Reply:
If you are talking about [this comment](https://github.com/python/cpython/pull/5691/files#r279120350).

I interpreted that discussion as " `str` and `bytes` could also be fed through `_os.fspath` ", which is what this PR does. 

Do you have a different interpretation of that comment?